state/protocol/badger/mutator_test.go

```diff
@@ -1207,100 +1207,94 @@ func TestExtendEpochTransitionWithoutCommit(t *testing.T) {
-			epoch2NewParticipant := unittest.IdentityFixture(unittest.WithRole(flow.RoleVerification))
-			epoch2Participants := append(participants, epoch2NewParticipant).Sort(order.Canonical)
+			// finalizing block 1 should trigger EECC
+			metricsMock.On("EpochEmergencyFallbackTriggered").Twice()
+			protoEventsMock.On("EpochEmergencyFallbackTriggered").Once()
 
-			// create the epoch setup event for the second epoch
-			epoch2Setup := unittest.EpochSetupFixture(
-				unittest.WithParticipants(epoch2Participants),
-				unittest.SetupWithCounter(epoch1Setup.Counter+1),
-				unittest.WithFinalView(epoch1FinalView+1000),
-				unittest.WithFirstView(epoch1FinalView+1),
-			)
+			// we begin the epoch in the EpochStaking phase and
+			// block 1 will be the first block on or past the epoch commitment deadline
+			block1 := unittest.BlockWithParentFixture(head)
+			block1.Header.View = epoch1CommitmentDeadline + rand.Uint64()%2
+			err = state.Extend(context.Background(), block1)
+			require.NoError(t, err)
+			err = state.Finalize(context.Background(), block1.ID())
+			require.NoError(t, err)
 
-			receipt1, seal1 := unittest.ReceiptAndSealForBlock(block1)
-			receipt1.ExecutionResult.ServiceEvents = []flow.ServiceEvent{epoch2Setup.ServiceEvent()}
+			assertEpochEmergencyFallbackTriggered(t, db)
 
-			// add a block containing a receipt for block 1
+			// block 2 will be the first block past the first epoch boundary
 			block2 := unittest.BlockWithParentFixture(block1.Header)
-			block2.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(receipt1)))
+			block2.Header.View = epoch1FinalView + 1
 			err = state.Extend(context.Background(), block2)
 			require.NoError(t, err)
 			err = state.Finalize(context.Background(), block2.ID())
 			require.NoError(t, err)
 
-			// block 3 seals block 1
-			block3 := unittest.BlockWithParentFixture(block2.Header)
-			block3.SetPayload(flow.Payload{
-				Seals: []*flow.Seal{seal1},
-			})
-			err = state.Extend(context.Background(), block3)
-			require.NoError(t, err)
-
-			// block 4 will be the first block for epoch 2
-			block4 := unittest.BlockWithParentFixture(block3.Header)
-			block4.Header.View = epoch1Setup.FinalView + 1
-
-			// inserting block 4 should trigger EECC
-			metricsMock.On("EpochEmergencyFallbackTriggered").Once()
-
-			err = state.Extend(context.Background(), block4)
-			require.NoError(t, err)
-
-			assertEpochEmergencyFallbackTriggered(t, db)
-
-			// epoch metrics should not be emitted
-			metricsMock.AssertNotCalled(t, "EpochTransition", epoch2Setup.Counter, mock.Anything)
-			metricsMock.AssertNotCalled(t, "CurrentEpochCounter", epoch2Setup.Counter)
+			// since EECC has been triggered, epoch transition metrics should not be updated
+			metricsMock.AssertNotCalled(t, "EpochTransition", mock.Anything, mock.Anything)
+			metricsMock.AssertNotCalled(t, "CurrentEpochCounter", epoch1Setup.Counter+1)
 			metricsMock.AssertExpectations(t)
+			protoEventsMock.AssertExpectations(t)
```

Comment:
I would suggest using [new mockery API](https://github.com/onflow/flow-go/blob/jordan/6227-eecc-trigger-condition/module/mock/compliance_metrics.go#L88) that automatically call `AssertExpectations`

Reply:
Done here https://github.com/onflow/flow-go/pull/2625/commits/38ff9cc69a0b0a8459b622fbf540afdfa424e0f2